### PR TITLE
test-bot: handle conflicted formulas and unlinked dependencies

### PR DESCRIPTION
**Pull request by ablyler**
these changes are primarily needed for php tap pull requests, as they usally contain multiple formulas depend on conflicting php versions:
- unlink any conflicted formulas
- link any dependencies that are installed and not linked

### CODE REVIEW

**Comment by mikemcquaid** [comment]
Can you give an example of a failing PR? Thanks!

**Comment by ablyler** [comment]
@mikemcquaid: Sure! https://github.com/Homebrew/homebrew-php/pull/1788

**Comment by ablyler** [comment]
Hey @mikemcquaid, don't mean to be a pain, but can you take a look at this?  The problem we have is with the bots and multiple php versions.  This would ensure that any conflicting formulas are unlinked and any dependancies are linked and fix the multiple php version issue we have in @Homebrew/php

**Comment by mikemcquaid** [comment]
@ablyler I'm on vacation

**Comment by ablyler** [comment]
@mikemcquaid no worries, I'll merge the pending homebrew-php pull requests for now.  can deal with bottles later.  have a good vacation, i'll stop bugging you now. :-)

File: Library/Homebrew/cmd/test-bot.rb [hunk 1]
```diff
@@ -428,9 +428,37 @@ def formula formula_name
...
+            test "brew", "unlink", "--force", conflict.name
+          end
+        end
+      end
+
       installed = Utils.popen_read("brew", "list").split("\n")
       dependencies = Utils.popen_read("brew", "deps", "--skip-optional",
                                       canonical_formula_name).split("\n")
+
+      # link any dependancies that are already installed
```
Comment (mikemcquaid) [inline Library/Homebrew/cmd/test-bot.rb]: And this one too, thanks.

File: Library/Homebrew/cmd/test-bot.rb [hunk 2]
```diff
@@ -428,9 +428,37 @@ def formula formula_name
         return
       end
 
+      # unlink any conflicts
```
Comment (mikemcquaid) [inline Library/Homebrew/cmd/test-bot.rb]: Please remove this comment, thanks.

File: Library/Homebrew/cmd/test-bot.rb [hunk 3]
```diff
@@ -428,9 +428,37 @@ def formula formula_name
...
+      formula.conflicts.each do |conflict|
+        confict_formula = Formulary.factory(conflict.name)
+
+        if confict_formula.installed? && confict_formula.linked_keg.exist?
+          test "brew", "unlink", "--force", conflict.name
+        end
```
Comment (mikemcquaid) [inline Library/Homebrew/cmd/test-bot.rb]: Maybe pull the inside of this loop into either a function or combine the blocks below by getting all the dependency formulae and the conflict formula into a single array.

File: Library/Homebrew/cmd/test-bot.rb [hunk 4]
```diff
@@ -428,9 +428,37 @@ def formula formula_name
...
+      end
+
       installed = Utils.popen_read("brew", "list").split("\n")
       dependencies = Utils.popen_read("brew", "deps", "--skip-optional",
                                       canonical_formula_name).split("\n")
+
+      # link any dependancies that are already installed
+      (installed & dependencies).each do |installed_dependency|
+        installed_dependency_formula = Formulary.factory(installed_dependency)
+        if installed_dependency_formula.installed? && !installed_dependency_formula.keg_only? && !installed_dependency_formula.linked_keg.exist?
```
Comment (mikemcquaid) [inline Library/Homebrew/cmd/test-bot.rb]: Can you add a newline in here somewhere? Thanks!

**Comment by ablyler** [comment]
@mikemcquaid updated per your feedback

**Comment by mikemcquaid** [comment]
Thanks @ablyler!

**Pull request closed without merging**